iAM-legacy/Simulation/SplitTreatmentLimit.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Simulation.Interface;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Simulation
+{
+    public class SplitTreatmentLimit : ISplitTreatmentLimit
+    {
+        public int Rank { get; }
+
+        public float Amount { get; }
+
+        public List<float> Percentages { get; }
+
+
+        public SplitTreatmentLimit(string rank, string amount, string percentages)
+        {
+            Rank = Convert.ToInt32(rank);
```

Comment:
Same here, consider making some private functions to handle the data operations or preferably some new classes.